fl4health/clients/basic_client.py

```diff
@@ -130,19 +130,27 @@ def get_parameters(self, config: Config) -> NDArrays:
         assert self.model is not None and self.parameter_exchanger is not None
         return self.parameter_exchanger.push_parameters(self.model, config=config)
 
-    def set_parameters(self, parameters: NDArrays, config: Config) -> None:
+    def set_parameters(self, parameters: NDArrays, config: Config, fitting_round: bool) -> None:
         """
         Sets the local model parameters transfered from the server using a parameter exchanger to coordinate how
-        parameters are set. If it's the first time the model is being initialized, we assume the full model is being
+        parameters are set. In the first fitting round, we assume the full model is being
         initialized and use the FullParameterExchanger() to set all model weights.
+        Otherwise, we use the appropriate parameter exchanger defined by the user depending on the
+        federated learning algorithm being used.
 
         Args:
             parameters (NDArrays): Parameters have information about model state to be added to the relevant client
                 model but may contain more information than that.
             config (Config): The config is sent by the FL server to allow for customization in the function if desired.
+            fitting_round (bool): Boolean that indicates whether the current federated learning round is a
+                fitting round or an evaluation round.
+                This is used to help determine which parameter exchange should be used for pulling parameters.
+                A full parameter exchanger is only used if the current federated learning round is the very
+                first fitting round.
         """
         assert self.model is not None
-        if not self.model_weights_initialized:
+        current_server_round = self.narrow_config_type(config, "current_server_round", int)
```

Comment:
Since we're no longer using `self.model_weights_initialized`, I think you can delete the property from `BasicClient`?

Reply:
Yes good catch. Sorry I thought I deleted it already but apparently I forgot.

Reply:
There is one place that this was actually being used `examples/warm_up_example/warmed_up_fenda/client.py` where we need to address this. I think there we can edit `get_model` and `initialize_all_model_weights` as follows to get the same effect:
```
def get_model(self, config: Config) -> nn.Module:
    return FendaModel(LocalCnn(), GlobalCnn(), FendaClassifier(FendaJoinMode.CONCATENATE)).to(self.device)
```
and
```
# Overriding the initialize_all_model_weights function is required to load the pretrained model
def initialize_all_model_weights(self, parameters: NDArrays, config: Config) -> None:
    # Load the pretrained model
    self.warmed_up_module.load_from_pretrained(self.model)
```

Reply:
I think this fix makes sense too. I've applied it in the latest commit.